numba/core/typing/typeof.py

```diff
@@ -94,6 +94,9 @@ def _typeof_type(val, c):
     if issubclass(val, np.generic):
         return types.NumberClass(numpy_support.from_dtype(val))
 
+    if issubclass(val, types.Type):
+        return types.TypeRef(val)
```

Comment:
It'd might be useful to make the repr on `TypeRef` more verbose? Using `class MyArray(types.Array): pass`, 
this:
```python
print(typeof(MyArray(types.float64, 3, 'C', True)))
```
gives
```
typeref[readonly array(float64, 3d, C)]
```
Maybe not for this PR though?

Reply:
i made an issue to track this: https://github.com/numba/numba/issues/7055

Reply:
thanks